thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/SubscriptionGroupManagerImpl.java

```diff
@@ -16,15 +16,36 @@
 import ai.startree.thirdeye.datalayer.dao.GenericPojoDao;
 import ai.startree.thirdeye.spi.datalayer.bao.SubscriptionGroupManager;
 import ai.startree.thirdeye.spi.datalayer.dto.SubscriptionGroupDTO;
+import com.codahale.metrics.CachedGauge;
+import com.codahale.metrics.MetricRegistry;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
+import java.util.concurrent.TimeUnit;
 
 @Singleton
 public class SubscriptionGroupManagerImpl extends
     AbstractManagerImpl<SubscriptionGroupDTO> implements SubscriptionGroupManager {
 
   @Inject
-  public SubscriptionGroupManagerImpl(GenericPojoDao genericPojoDao) {
+  public SubscriptionGroupManagerImpl(final GenericPojoDao genericPojoDao,
+      final MetricRegistry metricRegistry) {
     super(SubscriptionGroupDTO.class, genericPojoDao);
+    metricRegistry.register("subscriptionsCountTotal",
+        new CachedGauge<Integer>(15, TimeUnit.MINUTES) {
+      @Override
+      public Integer loadValue() {
+        return findAll().stream()
+            .filter(SubscriptionGroupManagerImpl::isPairPresent)
+            .map(sg -> sg.getAlertAssociations().size() * sg.getSpecs().size())
+            .reduce(0, Integer::sum);
+      }
+    });
+  }
+  
+  private static boolean isPairPresent(final SubscriptionGroupDTO subscriptionGroup) {
```

Comment:
I understand the function and then the name but the name in itself is super confusing. Like

```
            .filter(SubscriptionGroupManagerImpl::isPairPresent)
```
is not clear to be at all until I read the function code. In this case, I do not see the utility of the method anywhere else and I do not see it as a piece of independent computation.

What would have made sense to me if there was a method to compute the product and return that directly. However, in the current case I think having the logic in the method is way more explicit.

Reply:
Yeah agree, will dissolve the method in favour of chaining the filters.